js/pokemon.js

```diff
@@ -1146,3 +1208,30 @@ function loadState() {
         }
     }
 }
+
+Array.prototype.equals = function (array) {
```

Comment:
It's best practice not to add anything to native prototypes. Though tempting and likely to be fine here, the problem is if you use multiple libraries that try to make the same function on the same prototype. You're likely to end up with unexpected behaviour in both. Better to replace this function with something like arraysEqual(array1, array2).